perses/annihilation/new_relative.py

```diff
@@ -0,0 +1,1213 @@
+        sterics_addition += "softcore_alpha = %f;" % self.softcore_alpha
+        sterics_addition += "sigma = (1-lambda_sterics)*sigmaA + lambda_sterics*sigmaB;"
+        sterics_addition += "lambda_alpha = lambda_sterics*(1-lambda_sterics);"
+        return sterics_addition
+
+    def _nonbonded_custom_electrostatics_common(self):
+        """
+        Get a custom electrostatics expression that is common to all nonbonded methods
+
+        Returns
+        -------
+        electrostatics_addition : str
+            The common electrostatics energy expression
+        """
+        electrostatics_addition = "chargeprod = (1-lambda_electrostatics)*chargeprodA + lambda_electrostatics*chargeprodB;" #interpolation
+        electrostatics_addition += "reff_electrostatics = sqrt(softcore_beta*lambda_beta + r^2);" # effective softcore distance for electrostatics
+        electrostatics_addition += "softcore_beta = %f;" % (self.softcore_beta / self.softcore_beta.in_unit_system(unit.md_unit_system).unit)
+        electrostatics_addition += "ONE_4PI_EPS0 = %f;" % ONE_4PI_EPS0 # already in OpenMM units
+        electrostatics_addition += "lambda_beta = lambda_electrostatics*(1-lambda_electrostatics);"
+        return electrostatics_addition
+
+    def _nonbonded_custom_nocutoff(self):
+        """
+        Get a part of the nonbonded energy expression when there is no cutoff.
+
+        Returns
+        -------
+        sterics_energy_expression : str
+            The energy expression for U_sterics
+        electrostatics_energy_expression : str
+            The energy expression for electrostatics
+        """
+        # soft-core Lennard-Jones
+        sterics_energy_expression = "U_sterics = 4*epsilon*x*(x-1.0); x = (sigma/reff_sterics)^6;"
+        # soft-core Coulomb
+        electrostatics_energy_expression = "U_electrostatics = ONE_4PI_EPS0*chargeprod/reff_electrostatics;"
+        return sterics_energy_expression, electrostatics_energy_expression
+
+    def _nonbonded_custom_cutoff(self, epsilon_solvent, r_cutoff):
+        """
+        Get the energy expressions for sterics and electrostatics under a reaction field assumption.
+
+        Parameters
+        ----------
+        epsilon_solvent : float
+            The reaction field dielectric
+        r_cutoff : float
+            The cutoff distance
+
+        Returns
+        -------
+        sterics_energy_expression : str
+            The energy expression for U_sterics
+        electrostatics_energy_expression : str
+            The energy expression for electrostatics
+        """
+        # soft-core Lennard-Jones
+        sterics_energy_expression = "U_sterics = 4*epsilon*x*(x-1.0); x = (sigma/reff_sterics)^6;"
+
+        electrostatics_energy_expression = "U_electrostatics = ONE_4PI_EPS0*chargeprod*(reff_electrostatics^(-1) + k_rf*reff_electrostatics^2 - c_rf);"
```

Comment:
I'm pretty sure we need to force `c_rf = 0` for any interactions involving the alchemical atoms in order to not have reaction field generate garbage.

For atoms that remain in `standard_nonbonded_force`, we're probably fine.